### PR TITLE
#23 - Add endpoint to return element path

### PR DESCRIPTION
**Pull request by Ammarbr**
**What's in the PR**
* Add ElementPathHandler
* Add method to return all available paths for a given element

**How to test manually**
* See [#37](https://github.com/mig-frankfurt/dataelementhub.rest/pull/37)

### CODE REVIEW

File: src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  /**
+   * Complete a given path until the namespace.
+   */
+  public static List<String> completePaths(CloseableDSLContext ctx, int userId,
+      List<String> startPartialPaths) {
+    List<String> partialPaths = new ArrayList<String>(startPartialPaths);
+    for (int i = 0; i < partialPaths.size(); i++) {
+      String partialPath = partialPaths.get(i);
+      String urn;
+      if (partialPath.toLowerCase().contains("namespace")) {
```
Comment (michael-82) [inline src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java:48]: There are constants for the element types, e.g. ElementType.NAMESPACE.getLiteral() - better check against those. (with toUpperCase ofc)

File: src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        continue;
+      } else if (partialPath.contains("/")) {
```
Comment (michael-82) [inline src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java:50]: The else if branch is not needed. Just use urn = partialPath.split("/")[0]; - if the string does not contain the delimiter, it uses the whole string anyways.

File: src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            .getNamespaceIdentification(ctx, urn);
+        if (!DaoUtil.accessLevelGranted(namespaceIdentification.getIdentifier(),
+            userId, DaoUtil.READ_ACCESS_TYPES)) {
+          partialPaths.remove(partialPath);
+        } else {
+          partialPaths.add(namespaceIdentification.getNamespaceUrn() + "/" + partialPath);
+        }
+      }
+    }
+    boolean pathsCompleted = partialPaths.stream().allMatch(n -> n.contains("namespace"));
```
Comment (michael-82) [inline src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java:74]: can this be inlined?

File: src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    return element.getDefinitions().get(0).getDesignation();
+  }
+
+  /**
+   * Return list of scopedIdentifiers for all parent elements.
+   */
+  public static List<ScopedIdentifier> getParentScopedIdentifiers(
+      CloseableDSLContext ctx, String urn) {
+    int scopedIdentifierId = IdentificationHandler.getScopedIdentifier(ctx, urn).getId();
+    List<ScopedIdentifier> parentElements =
```
Comment (michael-82) [inline src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java:125]: this variable is redundand. just directly return the ctx.select...statement

Comment (Ammarbr) [reply src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java:125]: DONE

File: src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java (hunk 2 above)
Comment (Ammarbr) [reply src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java:50]: DONE

File: src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java (hunk 3 above)
Comment (Ammarbr) [reply src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java:74]: DONE

File: src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java (hunk 1 above)
Comment (Ammarbr) [reply src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java:48]: DONE

**Pull request merged**